pkg/lib/repo/local/pull.go

```diff
@@ -70,8 +70,16 @@ func (l *localRepo) PullModel(ctx context.Context, src oras.ReadOnlyTarget, ref
 		return fmt.Errorf("failed to get %s layer: %w", constants.FormatMediaTypeForUser(desc.MediaType), err)
 	}
 	var semErr error
+	// In some cases, manifests can contain duplicate digests. If we try to concurrently pull the same digest
+	// twice, a race condition will cause the pull the fail.
+	pulledDigests := map[string]bool{}
```

Comment:
this can be a preallocated map 
```Go
	pulledDigests := make(map[string]bool, len(toPull))
```

Reply:
It could, but in the case we're trying to solve this would be overallocating -- this is a set used to check whether we've seen each element before. With v0.4.0, we can create ModelKits that have the same layer dozens of times (e.g. empty `readme.md` files in different directories)